### PR TITLE
upgrade: Go version 1.22.4 for vulnerability GO-2024-2887

### LINKED ISSUE #59: Go Vulnerability Report
## Go Vulnerability Report

The following vulnerabilities were identified in the project:

```
=== Symbol Results ===

Vulnerability #1: GO-2024-2887
    Unexpected behavior from Is methods for IPv4-mapped IPv6 addresses in
    net/netip
  More info: https://pkg.go.dev/vuln/GO-2024-2887
  Standard library
    Found in: net/netip@go1.22.3
    Fixed in: net/netip@go1.22.4
    Example traces found:
      #1: aws.go:7:2: logger.init calls http.init, which eventually calls netip.Addr.IsLoopback
      #2: aws.go:7:2: logger.init calls http.init, which eventually calls netip.Addr.IsMulticast

Your code is affected by 1 vulnerability from the Go standard library.
This scan also found 0 vulnerabilities in packages you import and 1
vulnerability in modules you require, but your code doesn't appear to call these
vulnerabilities.
Use '-show verbose' for more details.
```

### PR DESCRIPTION
**Pull request by mmcgehee-ccc**
closes #59

### CODE REVIEW

File: .github/workflows/security-scan.yml [hunk 1]
```diff
@@ -1,14 +1,17 @@
 name: Security Scan
 
 on:
+  push:
+    branches:
+      - upgrade/go-1-22-4
```
Comment (jwatson-ccc) [inline .github/workflows/security-scan.yml:6]: Do we want to remove this?

Comment (jwatson-ccc) [reply .github/workflows/security-scan.yml:6]: Looks like you are still testing. Other then this, everything looks good.

Comment (mmcgehee-ccc) [reply .github/workflows/security-scan.yml]: Yeah, sorry I should've removed the review request when I decided to do further testing.

**Review by jwatson-ccc — APPROVED** [review]
LGTM

**Pull request merged**